### ISSUE TITLE
Android screenshot layer settings don't work

### ISSUE THREAD

**Issue by lunarpapillo** [LunarG/VulkanTools#965]
The code to extract screenshot layer settings in Android uses the `getprop` command to read back values for `debug.vulkan.screenshot` and `debug.vulkan.screenshot.dir`.  But the code itself returns values in a static character array:
```
        static char android_env[64] = {};

        char *android_exec(const char *cmd) {
            FILE *pipe = popen(cmd, "r");
            if (pipe != nullptr) {
                fgets(android_env, 64, pipe);
                pclose(pipe);
            }

            // Only if the value is set will we get a string back
            if (strlen(android_env) > 0) {
                __android_log_print(ANDROID_LOG_INFO, "screenshot", "Vulkan screenshot layer capturing: %s", android_env);
                // Do a right strip of " ", "\n", "\r", "\t" for the android_env string
                string android_env_str(android_env);
                android_env_str.erase(android_env_str.find_last_not_of(" \n\r\t") + 1);
                snprintf(android_env, sizeof(android_env), "%s", android_env_str.c_str());
                return android_env;
            }

            return nullptr;
        }

        char *android_getenv(const char *key) {
            std::string command("getprop ");
            command += key;
            return android_exec(command.c_str());
        }
```
This means that the same buffer is returned for each query, which effectively causes all settings to become equal to the last setting queried.

An excerpt from a `logcat` is illustrative.  After:
```
setprop debug.vulkan.screenshot 0,89,185
setprop debug.vulkan.screenshot.dir /mnt/sdcard
```
the `logcat` shows:
```
        03-05 01:01:32.640 24567 24675 I vulkan  : Loaded layer VK_LAYER_LUNARG_screenshot
        03-05 01:01:32.657 24567 24675 I screenshot: Vulkan screenshot layer capturing:
        03-05 01:01:32.670 24567 24675 I screenshot: Vulkan screenshot layer capturing: /mnt/sdcard
        03-05 01:01:32.683 24567 24675 I screenshot: Vulkan screenshot layer capturing: 0,89,185
        03-05 01:01:33.004 24567 24675 I screenshot: Screen capture file is: 0,89,185/0.ppm
        03-05 01:01:33.011 24567 24675 D screenshot: Failed to open output file: 0,89,185/0.ppm.  Be sure to grant read and write permissions.
        03-05 01:01:35.669 24567 24675 I screenshot: Screen capture file is: 0,89,185/89.ppm
        03-05 01:01:35.679 24567 24675 D screenshot: Failed to open output file: 0,89,185/89.ppm.  Be sure to grant read and write permissions.
        03-05 01:01:38.047 24567 24675 I screenshot: Screen capture file is: 0,89,185/185.ppm
        03-05 01:01:38.272 24567 24675 D screenshot: Failed to open output file: 0,89,185/185.ppm.  Be sure to grant read and write permissions.
```
It's clear in this case that the value of the `debug.vulkan.screenshot` property is also being used as `debug.vulkan.screenshot.dir`.

Also, the `screenshot_layer.md` documentation is incorrect; it refers to using `debug.vulkan.dir` to set the directory where screenshots will be saved.  The correct value is `debug.vulkan.screenshot.dir`.

**Comment by lunarpapillo** [comment]
BTW, this bug prevents correct `gfxreconstruct` testing on Android.

**Comment by davidlunarg** [comment]
Fixed in commit f1ed7a325ed9d66d9121d0bce875d7017339a27f a long, long time ago.
